File: CS/FileManager_Example/Controllers/FMSettingsHelper.cs
﻿using DevExpress.Web;

namespace FileManager_Example.Helpers {
    public class FMSettingsHelper {
        public static DevExpress.Web.Mvc.FileManagerSettings CreateFileManagerDownloadSettings() {
            var settings = new DevExpress.Web.Mvc.FileManagerSettings();
            var editingSettings = CreateFileManagerEditingSettings();
            settings.SettingsEditing.Assign(editingSettings);
            settings.Name = "FileManager";
            return settings;
        }
        public static FileManagerSettingsEditing CreateFileManagerEditingSettings() {
            return new FileManagerSettingsEditing(null) {
                AllowCreate = true,
                AllowMove = true,
                AllowDelete = true,
                AllowRename = true,
                AllowCopy = true,
                AllowDownload = true
            };
        }
    }
}

File: VB/FileManager_Example/Controllers/FMSettingsHelper.vb
Imports DevExpress.Web

Namespace FileManager_Example.Helpers
	Public Class FMSettingsHelper
		Public Shared Function CreateFileManagerDownloadSettings() As DevExpress.Web.Mvc.FileManagerSettings
			Dim settings = New DevExpress.Web.Mvc.FileManagerSettings()
			Dim editingSettings = CreateFileManagerEditingSettings()
			settings.SettingsEditing.Assign(editingSettings)
			settings.Name = "FileManager"
			Return settings
		End Function
		Public Shared Function CreateFileManagerEditingSettings() As FileManagerSettingsEditing
			Return New FileManagerSettingsEditing(Nothing) With {
				.AllowCreate = True,
				.AllowMove = True,
				.AllowDelete = True,
				.AllowRename = True,
				.AllowCopy = True,
				.AllowDownload = True
			}
		End Function
	End Class
End Namespace